components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/listener/ProvisioningIdentityProviderMgtListener.java

```diff
@@ -56,6 +56,23 @@ public boolean doPreDeleteIdP(String idPName, String tenantDomain) throws Identi
         return true;
     }
 
+    /**
+     * Clear Provisioning Connector Cache before deleting IDPs.
+     *
+     * @param tenantDomain Tenant domain to delete IdPs.
+     * @return
+     * @throws IdentityProviderManagementException
+     */
+    @Override
+    public boolean doPreDeleteIdPs(String tenantDomain) throws IdentityProviderManagementException {
+
+        ProvisioningConnectorCache.getInstance().clear();
```

Comment:
This affects to all the tenants. Lets skip removing from  ProvisioningConnectorCache